### PR TITLE
Fix Tooltip on ScatterChart with allowDuplicateCategory=false

### PR DESCRIPTION
**Pull request by PavelVanecek**
## Description

The logic in this method is getting convoluted again but it works ... for now. If it turns out to be problematic I will break it down further.

## Related Issue

Fixes https://github.com/recharts/recharts/issues/5546

Reported in https://github.com/recharts/recharts/discussions/5545

### CODE REVIEW

File: src/state/selectors/selectors.ts [hunk 1]
```diff
@@ -209,7 +210,12 @@ export const combineTooltipPayload = (
     // BaseAxisProps does not support nameKey but it could!
     const finalNameKey: DataKey<any> | undefined = settings?.nameKey; // ?? tooltipAxis?.nameKey;
     let tooltipPayload: unknown;
-    if (tooltipAxis?.dataKey && !tooltipAxis?.allowDuplicatedCategory && Array.isArray(sliced)) {
+    if (
```
Comment (PavelVanecek) [inline src/state/selectors/selectors.ts:213]: Too much coincidence in one `if` for my liking. But the storybook works well and all tests are passing so

**Review by ckifer — APPROVED** [review]
Always edge cases I guess. Thanks for looking at this
